jsapp/js/projects/projectsTable/colResize.tsx

```diff
@@ -0,0 +1,271 @@
+) {
+  switch (action.type) {
+    case 'resize':
+      return {
+        ...columnWidths,
+        [action.column]: action.width,
+      };
+    default:
+      return {...columnWidths};
+  }
+}
+
+/**
+ * The minimum resizing width for a given column.
+ *
+ * This prevents columns from being entirely cropped away.
+ * Currently a minimum of 108px.
+ * Can be hardcoded per-field.
+ */
+function minColWidth(fieldname: string) {
+  const defaultMinimumWidth = 108;
+  const minimums: ColumnWidths = {
+    countries: 116,
+    dateModified: 144,
+    dateDeployed: 144,
+  };
+  return minimums[fieldname] || defaultMinimumWidth;
+}
+/**
+ * The maximum resizing width for a given column.
+ *
+ * This is intended to reduce confusion from resizing 1 column absurdly wide.
+ * Currently an arbitrary maximum 800px for all fields.
+ * Can be hardcoded per-field.
+ */
+function maxColWidth(fieldname: string) {
+  const defaultMaximumWidth = 800;
+  const maximums: ColumnWidths = {};
+  return maximums[fieldname] || defaultMaximumWidth;
+}
+/**
+ * A helper to clamp between the minimum and maximum allowed resizing widths.
+ */
+function clampedColumnWidth(fieldname: string, width: number) {
+  return Math.min(
+    Math.max(width, minColWidth(fieldname)),
+    maxColWidth(fieldname)
+  );
+}
+
+/**
+ * <ColumnResizer/>
+ *
+ * Creates and updates styles for column widths based on drag events on resize
+ * handles.
+ */
+export default function ColumnResizer() {
+  // State used in rendering new styles
+  const [columnWidths, dispatch] = useReducer(
+    columnWidthReducer,
+    {} as ColumnWidths
+  );
+  // Polish. This could perhaps be merged into a single state with isDragging
+  const [shouldRenderBodyCursor, setShouldRenderBodyCursor] = useState(false);
+
+  // Refs, for use by event handlers. They won't trigger re-renders.
+  /** Bool: Have we started dragging a resize handle? */
+  const isDraggingRef = useRef(false);
+  /** The field name of the column being resized. */
+  const draggingColumnRef: React.MutableRefObject<string | null> = useRef(null);
+  /** The pageX when we started interacting with the resize handler */
+  const dragStartXRef: React.MutableRefObject<number | null> = useRef(null);
+  /** The width of the header cell when we initially touched its resize handler */
+  const dragStartWidthRef: React.MutableRefObject<number | null> = useRef(null);
+  /** The pageX at the previous mousedown/mousemove.
+      Used for de-duping mousemove */
+  const dragPrevXRef = useRef(0);
+  /** The column width we used in the previous mousedown/mousemove.
+      Used for de-duping mouse renders. */
+  const dragPrevWidthRef = useRef(0);
+
+  /**
+   * The singular DOM event handler for column resizing.
+   * One consolidated function. It switches on event.type, delegates when
+   * necessary, and exits early.
+   */
+  const handlerRef = useRef((e: Event) => {
+    // TypeScript guard
+    if (e instanceof MouseEvent) {
```

Comment:
Could we do this instead of wrapping whole function in `if` statement?

```ts
if (e instanceof MouseEvent === false) {
  return;
}
```

This way the guard is here on top, and the rest of the code doesn't need to be indentend so much :)

Reply:
Oh, yep, if it works!